client/signup/controller.js

```diff
@@ -223,25 +223,43 @@ export default {
 		store.set( 'signup-locale', localeFromParams );
 
 		const isOnboardingFlow = flowName === 'onboarding';
-		if ( isOnboardingFlow ) {
+		if ( isOnboardingFlow && ! context.querystring?.includes( 'redirected=true' ) ) {
```

Comment:
```suggestion
		if ( isOnboardingFlow && ! context.querystring?.includes( 'redirected=true' ) ) {
```

Maybe use a more unique param name `redirected_1220=true`, so it will be easier to search for this code if anything goes wrong or someone is just wondering why there is an extra redirect.

Reply:
Good idea!